service/lambda/api_op_AddLayerVersionPermission.go

```diff
@@ -208,6 +210,54 @@ func (m *opAddLayerVersionPermissionResolveEndpointMiddleware) HandleSerialize(c
 		)
 	}
 
+	authSchemes, err := internalauth.GetAuthenticationSchemes(&resolvedEndpoint.Properties)
+	if err != nil {
+		var nfe *internalauth.NoAuthenticationSchemesFoundError
+		if errors.As(err, &nfe) {
+			// if no auth scheme is found, default to sigv4
+			signingName := "lambda"
+			signingRegion := m.BuiltInResolver.(*BuiltInResolver).Region
+			ctx = awsmiddleware.SetSigningName(ctx, signingName)
+			ctx = awsmiddleware.SetSigningRegion(ctx, signingRegion)
+
+		}
+		authSchemes = []internalauth.AuthenticationScheme{}
+	}
+
+	for _, authScheme := range authSchemes {
+		switch v := authScheme.(type) {
+		case *internalauth.AuthenticationSchemeV4:
+			v4Scheme, _ := authScheme.(*internalauth.AuthenticationSchemeV4)
+			var signingName, signingRegion string
+			if v4Scheme.SigningName == nil {
```

Comment:
**comment**: If these are always the default (and I believe they are) we don't really need a no auth schemes found error/branch. We can just always set the defaults. 

e.g.

```go

ctx = awsmiddleware.SetSigningName(ctx, "lambda")
ctx = awsmiddleware.SetSigningRegion(ctx,  signingRegion)

authSchemes, err := internalauth.GetAuthenticationSchemes(...)
if err != nil { 
     return err
}

for _, authScheme := rangeAuthSchemes {
     switch v := authScheme.(type) {
            ...
     }
}

```

Reply:
yah these are always the defaults. i believe the no auth schemes found error is part of the stated requirements.